skyfield/keplerlib.py

```diff
@@ -523,74 +544,74 @@ def kepler(x):
     lower = zeros_like(dt, dtype='float64')
     oldx = zeros_like(dt, dtype='float64')
 
-    lower[past] = x[past]
-    upper[future] = x[future]
+    copyto(lower, x, where=past)
+    copyto(upper, x, where=future)
 
     while (kfun[past] > dt[past]).any():
-        upper[past] = lower[past]
+        copyto(upper, lower, where=past)
         lower[past] *= 2
-        oldx[past] = x[past]
-        x[past] = bracket(lower[past], -bound, bound)
+        copyto(oldx, x, where=past)
+        orb_ind = sum(past, axis=1)
+        x[past] = clip(lower[past], repeat(-bound, orb_ind), repeat(bound, orb_ind))
         if (x[past] == oldx[past]).any():
             raise ValueError('The input delta time (dt) has a value of {0}.'
                              'This is beyond the range of DT for which we '
                              'can reliably propagate states. The limits for '
                              'this GM and initial state are from {1}'
                              'to {2}.'.format(dt, kepler(-bound), kepler(bound)))
-        kfun[past] = kepler(x[past])
+        kfun[past] = kepler_1d(x[past], orb_ind)
 
     while (kfun[future] < dt[future]).any():
-        lower[future] = upper[future]
+        copyto(lower, upper, where=future)
         upper[future] *= 2
-        oldx[future] = x[future]
-        x[future] = bracket(upper[future], -bound, bound)
+        copyto(oldx, x, where=future)
+        orb_ind = sum(future, axis=1)
+        x[future] = clip(upper[future], repeat(-bound, orb_ind), repeat(bound, orb_ind))
         if (x[future] == oldx[future]).any():
             raise ValueError('The input delta time (dt) has a value of {0}.'
                              'This is beyond the range of DT for which we '
                              'can reliably propagate states. The limits for '
                              'this GM and initial state are from {1} '
                              'to {2}.'.format(dt, kepler(-bound), kepler(bound)))
-        kfun[future] = kepler(x[future])
+        kfun[future] = kepler_1d(x[future], orb_ind)
 
-    x = amin(array([upper, amax(array([lower, (lower+upper)/2]), axis=0)]), axis=0)
+    x = copy(upper)
+    copyto(x, (upper+lower)/2, where=(lower<=upper))
```

Comment:
It took me a while, but I figured out that
```upper if lower>upper else (lower+upper)/2```
is equivalent to
```min(upper, max(lower, (lower+upper)/2))```
This also occurs further down.